lib/src/core/message/components/message_component.dart

```diff
@@ -99,8 +99,8 @@ abstract class IMessageComponent {
   /// The [ComponentType]
   ComponentType get type;
 
-  /// The custom id of this component, set by the user.
-  String get customId;
+  /// The custom id of this component set by the user, if any.
+  String? get customId;
```

Comment:
Add a TODO to remind us to make it nullable in the next major release as well please